river/tree/extremely_fast_decision_tree.py

```diff
@@ -136,23 +138,26 @@ def _new_learning_node(self, initial_stats=None, parent=None):
             depth = parent.depth + 1
 
         if self._leaf_prediction == self._MAJORITY_CLASS:
-            return EFDTLearningNodeMC(initial_stats, depth, self.splitter)
+            return EFDTLeafMajorityClass(initial_stats, depth, self.splitter)
         elif self._leaf_prediction == self._NAIVE_BAYES:
-            return EFDTLearningNodeNB(initial_stats, depth, self.splitter)
+            return EFDTLeafNaiveBayes(initial_stats, depth, self.splitter)
         else:  # NAIVE BAYES ADAPTIVE (default)
-            return EFDTLearningNodeNBA(initial_stats, depth, self.splitter)
+            return EFDTLeafNaiveBayesAdaptive(initial_stats, depth, self.splitter)
 
-    def _new_split_node(
-        self, split_test, target_stats=None, depth=0, existing_splitters=None
-    ):
+    def _branch_selector(
```

Comment:
I will change that once we decide upon the final convention. It could be either `_new_splitter` (but `splitter` is also used to refer to the attribute observers, so maybe it's better to use `_new_split`) or `_new_branch`.

Reply:
My bad, `_branch_selector` does a different job than the previous `_new_split_node`. I got confused :P

I discuss the functionality of this method in another comment.